### PR TITLE
Upgrade curator to v5.2

### PR DESCRIPTION
**Pull request by josefkarasek**

### CODE REVIEW

**Comment by josefkarasek** [comment]
Temporarily including certs and config files - in case you wanna try to run in or want to see my test env

**Comment by josefkarasek** [comment]
@jcantrill @richm I changed my mind about the use of the upstream configuration.
Let's continue using our custom config and call the convert script during each start of the container.
Our config is much simpler and also has the `.default` feature, which prevents people from deleting `.kibana` and `.searchguard` indices. 

Check out the readme, I think it will be clear to you what bothers me about the upstream config (the actions file)

File: curator/src/convert.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        actions_yaml = None
+        with open(self.actions_file, 'r') as f:
+            actions_string = f.read()
+            actions_yaml = yaml.load(actions_string, Loader=yaml.Loader)
+        if actions_yaml is None or actions_yaml['actions'] is None:
+            return True
+        else:
+            return False
+
+    def format_regex(self, project):
```
Comment (jcantrill) [inline curator/src/convert.py]: Do we properly handle other indices like '.operations', '.orphanted?

**Comment by richm** [comment]
is it necessary to use ruamel yaml instead of the default PyYAML that comes with the base rhel7?

**Comment by josefkarasek** [comment]
@richm PyYAML doesn't respect quotation marks. I had to use `SingleQuotedScalarString` from ruamel yaml.

I checked and ruamel is shipped in rhel7. @ewolinetz pls correct me if I'm wrong.

File: curator/src/convert.py (hunk 1 above)
Comment (josefkarasek) [reply curator/src/convert.py]: * `.operations` - yes
* `.orphanted` - I have no idea. How are these indices created? Do they contain the date of creation? I think that they should. Then they fit the `project.startswith('.') ` branch.
AFAIK only `.kibana` index 
```
project.kube-service-catalog.a5c2d3ed-121e-11e8-be27-0ebf08f2fae0.2018.02.15
project.logging.32004c38-121e-11e8-be27-0ebf08f2fae0.2018.02.15
.kibana
.operations.2018.02.15
.searchguard.logging-es-data-master-kkrxvdkl
```

See example result:
```
actions:
  1:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: false
      kind: regex
      filtertype: pattern
      value: '^project\.myapp\..*$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 7
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
  2:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: false
      kind: regex
      filtertype: pattern
      value: '^\.operations\..*$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 56
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
  3:
    action: delete_indices
    description: auto-generated on 2018-02-14 11:11
    filters:
    - exclude: true
      kind: regex
      filtertype: pattern
      value: '^project\.myapp\..*$|^\.operations\..*$|^\.searchguard\..*$|^\.kibana$'
    - direction: older
      filtertype: age
      source: name
      timestring: '%Y.%m.%d'
      unit_count: 30
      unit: days
    options:
      continue_if_exception: false
      timeout_override: '300'
      ignore_empty_list: true
```

**Comment by richm** [comment]
> I checked and ruamel is shipped in rhel7. @ewolinetz pls correct me if I'm wrong.

It is in the RHAOS channels, so we should be able to use that to build the curator image.

I could not find it in rhel7 or extras or optional.  It is in EPEL7, which we cannot use downstream.

**Comment by josefkarasek** [comment]
@richm yes, rhaos. Sorry for misleading info.

**Comment by richm** [comment]
the CI test failures are in the curator test, so maybe not a flake

File: curator/src/convert.py (hunk 1 above)
Comment (josefkarasek) [reply curator/src/convert.py]: @jcantrill please correct me if I'm wrong

**Comment by josefkarasek** [comment]
@richm should the test assume that curator can still run as a deployment and that config can be supplied as a config map? 

https://github.com/openshift/origin-aggregated-logging/pull/944/files#diff-2906f4cbcd037629ad1a6021dbae54fdR144

File: curator/README.md [hunk 2]
```diff
@@ -108,21 +113,10 @@ configuration in the deployed configmap:
 
     $ oc edit configmap/logging-curator
 
-If this does not redeploy automatically, redeploy manually:
-
-    $ oc deploy --latest logging-curator
```
Comment (richm) [inline curator/README.md]: oc rollout latest dc/logging-curator
oc rollout status -w dc/logging-curator

File: curator/README.md [hunk 3]
```diff
@@ -108,21 +113,10 @@ configuration in the deployed configmap:
...
     $ oc create configmap logging-curator --from-file=config.yaml=/path/to/mycuratorconfig.yaml ; sleep 1
 
 Then redeploy as above.
-
-You can also specify default values for the run hour, run minute, and age in
-days of the indices when processing the curator template.  Use
-`CURATOR_RUN_HOUR` and `CURATOR_RUN_MINUTE` to set the default runhour and
-runminute, `CURATOR_RUN_TIMEZONE` to set the run timezone, and use
-`CURATOR_DEFAULT_DAYS` to set the default index age in days. These are only
-used if not specified in the config file.
```
Comment (richm) [inline curator/README.md]: Do we still support these env. vars?  If not, why?

File: curator/src/convert.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+from parser import Parser
+from util import create_logger
+
+class LegacyConfigConverter():
+
+    LEGACY_CONFIG = 'CURATOR_CONF_LOCATION'
+    ACTIONS_FILE  = 'CURATOR_ACTIONS_FILE'
+    PROJ_PREFIX   = 'project.'
+    ACTIONS_FILE_WRITE_LOCATION = '/opt/app-root/src/actions.yaml'
```
Comment (richm) [inline curator/src/convert.py]: suggest using `os.getenv('HOME', '/opt/app-root/src') + '/actions.yaml'`

File: curator/README.md (hunk 2 above)
Comment (richm) [reply curator/README.md]: ah, it's a cronjob - never mind

File: curator/README.md (hunk 3 above)
Comment (richm) [reply curator/README.md]: Is there some way we can "translate" these into the cronjob settings?

**Comment by richm** [comment]
> @richm should the test assume that curator can still run as a deployment

No, since it is a cronjob.  If it is easy to abstract that out, rather than having to hardcode `dc` or `cronjob` in the test, that will make it easier to backport test fixes to 3.9 and earlier releases.  Otherwise, it will be tricky if we fix something in curator that we need to backport.

> and that config can be supplied as a config map?

How is config supplied now?

Are there changes to openshift-ansible for installation and upgrade?

File: curator/README.md (hunk 3 above)
Comment (josefkarasek) [reply curator/README.md]: They're not used by curator or the associated scripts anymore.

Instead the playbook uses them to generate the cronjob k8s object:
https://github.com/openshift/openshift-ansible/pull/7047/files#diff-f4c6ebdf7da2e4a833073d1f89e08221R164

**Comment by josefkarasek** [comment]
@richm configuration is supplied as a config map now. The test assumes it can be either config map or a secret. I suggest we expect only config maps.

There are many changes to openshift-ansible https://github.com/openshift/openshift-ansible/pull/7047/files#diff-f4c6ebdf7da2e4a833073d1f89e08221R164

**Comment by josefkarasek** [comment]
@jcantrill @ewolinetz  tests failing on `oc get cronjob logging-curator ` now.
I guess we need the ansible PR merged first

**Comment by ewolinetz** [comment]
Ansible pr: https://github.com/openshift/openshift-ansible/pull/7047

**Comment by ewolinetz** [comment]
@josefkarasek I think we need to update the deployment test scripts since we don't have a curator deploymentconfig

**Comment by ewolinetz** [comment]
@josefkarasek can you update this to be opened against the es5.x branch instead of master?

**Comment by richm** [comment]
@josefkarasek please rebase this patch on top of the es5.x branch and submit a PR against the es5.x branch

**Comment by josefkarasek** [comment]
Closing in favor of https://github.com/openshift/origin-aggregated-logging/pull/1020

**Comment by josefkarasek** [comment]
Closing in favor of https://github.com/openshift/origin-aggregated-logging/pull/1020

**Pull request closed without merging**